### PR TITLE
Pivot overflow: fix scrollbars appearing when the window is just barely wide enough to show all tabs

### LINKED ISSUE #15292: Pivot/Tabs overflow causes scrollbar to appear just before overflow menu appears
The Pivot/Tabs overflow menu causes scrollbar to appear just before overflow menu appears

Repro steps:
1. Open this example codesandbox of a pivot with overflow: https://codesandbox.io/s/pivot-overflow-menu-demo-grrgg (Standalone: https://grrgg.csb.app/)
2. Start with the window wide enough to show all tabs
3. Resize the window down until just *before* the overflow menu [...] appears.
4. Notice that the scrollbar appears at the bottom of the window

image

### PR DESCRIPTION
**Pull request by behowell**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #15292
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Pivot overflow: fix scrollbars appearing when the window is just barely wide enough to show all tabs

Set the overflow button to have `position: absolute; right: 0;` when it is hidden. This causes it to not take up layout space while hidden, but still have a size that can be measured by the `useOverflow` algorithm.

#### Focus areas to test

Pivot overflow

### CODE REVIEW

File: packages/react-tabs/src/components/Pivot/Pivot.styles.ts
Comment (behowell) [inline packages/react-tabs/src/components/Pivot/Pivot.styles.ts:222]: FYI @czearing, you will need to do this TODO when merging #15216. I would normally wait until after your PR is merged for this change (and not change it myself), but it looks like your PR is being held back for now. It should look like this after the TODO:
```js
overflowMenuButton: [
  classNames.overflowMenuButton,
  ...getLinkStyles(props, classNames),
  {
    visibility: 'hidden',
    position: 'absolute',
    right: 0,
  },
],
```

**Comment by xugao** [comment]
> Thanks for submitting this change, but due to work currently in progress to prepare `master` for our version 8 beta release, we're asking contributors to either wait a couple weeks to submit changes (if it's not urgent) or submit to the new `7.0` branch (if it's urgent). See #15222 for more details. Thank you for your contribution and understanding!

@behowell - you can ignore this comment :) your change is v8 related so it should be in `master`

File: packages/react-tabs/src/components/Pivot/Pivot.styles.ts
Comment (xugao) [inline packages/react-tabs/src/components/Pivot/Pivot.styles.ts:215]:
```suggestion
        ['button&']: getLinkStyles(props, classNames),
```

File: packages/react-tabs/src/components/Pivot/Pivot.styles.ts
Comment (xugao) [inline packages/react-tabs/src/components/Pivot/Pivot.styles.ts:222]:
```suggestion
        ['span&']: 
          {
            visibility: 'hidden',
            position: 'absolute',
            right: 0,
          },
```

**Review by xugao — APPROVED** [review]

File: packages/react-tabs/src/components/Pivot/Pivot.styles.ts
Comment (behowell) [reply packages/react-tabs/src/components/Pivot/Pivot.styles.ts:215]: I had left it like this to make it clearer how to remove this workaround when Button is fixed, but it's probably not necessary :)

File: packages/react-tabs/src/components/Pivot/Pivot.base.tsx
Comment (behowell) [inline packages/react-tabs/src/components/Pivot/Pivot.base.tsx:267]: It turns out that my original workaround didn't correctly apply hover visuals when when the links were displayed as tabs. This new workaround is a more straightforward fix, which applies different classes to the wrapper `<span>` and the inner `<button>`.

@czearing FYI with this change, you'll now need to add both classes (`link` and `overflowMenuButton`) to the button once  you remove the span.

**Pull request merged**
